Tasks/MavenV3/mavenutil.ts

```diff
@@ -193,64 +188,73 @@ interface RepositoryInfo {
-    return convertXmlStringToJson(pomContents).then(function (pomJson) {
+async function collectFeedRepositories(pomContents:string): Promise<any> {
+    return convertXmlStringToJson(pomContents).then(async function (pomJson) {
         let repos:RepositoryInfo[] = [];
         if (!pomJson) {
             tl.debug('Incomplete pom: ' + pomJson);
             return Q.resolve(repos);
         }
-        let collectionUrl = tl.getVariable("System.TeamFoundationCollectionUri");
-        return locationHelpers.assumeNuGetUriPrefixes(collectionUrl).then(function (packageUrl) {
-            tl.debug('collectionUrl=' + collectionUrl);
-            tl.debug('packageUrl=' + packageUrl);
-            let collectionHostname:string = url.parse(collectionUrl).hostname.toLowerCase();
-            let packageHostname:string = packageUrl[1];
-            if (packageHostname) {
-                url.parse(packageHostname).hostname.toLowerCase();
-            } else {
-                packageHostname = collectionHostname;
-            }
-            let parseRepos:(project) => void = function(project) {
-                if (project && project.repositories) {
-                    for (let r of project.repositories) {
-                        r = r instanceof Array ? r[0] : r;
-                        if (r.repository) {
-                            for (let repo of r.repository) {
-                                repo = repo instanceof Array ? repo[0] : repo;
-                                let url:string = repo.url instanceof Array ? repo.url[0] : repo.url;
-                                if (url && (url.toLowerCase().includes(collectionHostname) ||
-                                            url.toLowerCase().includes(packageHostname))) {
-                                tl.debug('using credentials for url: ' + url);
-                                repos.push({
-                                    id: (repo.id && repo.id instanceof Array)
-                                        ? repo.id[0]
-                                        : repo.id
-                                    });
-                                }
+        const collectionUrl = tl.getVariable("System.TeamFoundationCollectionUri");
+        let packagingLocation: pkgLocationUtils.PackagingLocation;
+        try {
+            packagingLocation = await pkgLocationUtils.getPackagingUris(pkgLocationUtils.ProtocolType.Maven);
+        } catch (error) {
+            tl.debug("Unable to get packaging URIs, using default collection URI");
+            tl.debug(JSON.stringify(error));
+            packagingLocation = {
+                PackagingUris: [collectionUrl],
+                DefaultPackagingUri: collectionUrl};
+        }
+
+        let packageUrl = packagingLocation.DefaultPackagingUri;
+        tl.debug('collectionUrl=' + collectionUrl);
+        tl.debug('packageUrl=' + packageUrl);
+        let collectionHostname:string = url.parse(collectionUrl).hostname.toLowerCase();
```

Comment:
Are we sure that it's right to use a `hostname` anymore? `https://dev.azure.com/kellie-test-project` would have the same hostname as `https://dev.azure.com/mseng`, correct?

Reply:
Good point. I'll look into how I can make this more resilient.